HidPkg/Crates/HiiKeyboardLayout/src/lib.rs

```diff
@@ -0,0 +1,782 @@
+//! HiiKeyboardLayout - provides support for UEFI HII Keyboard Layout structures
+//!
+//! This crate provides a rust wrapper around UEFI HII Keyboard Layout structures. The relevant structures defined in
+//! the UEFI spec are not well-suited for direct definition in rust; so this crate defines analogous rust structures and
+//! provides serialization/deserialization support for converting the rust structures into byte buffers and vice versa.
+//!
+//! ## Examples and Usage
+//!
+//! Retrieving a default (en-US) layout, writing it to a buffer, and then reading the buffer back into a rust structure:
+//!```
+//! use hii_keyboard_layout::{get_default_keyboard_pkg, HiiKeyboardPkg};
+//! use scroll::{Pread, Pwrite};
+//! let mut buffer = [0u8; 4096];
+//!
+//! let package = get_default_keyboard_pkg();
+//! buffer.pwrite(&package, 0).unwrap();
+//!
+//! let package2: HiiKeyboardPkg = buffer.pread(0).unwrap();
+//! assert_eq!(package, package2);
+//!```
+//!
+//! ## License
+//!
+//! Copyright (c) Microsoft Corporation. All rights reserved.
+//! SPDX-License-Identifier: BSD-2-Clause-Patent
+//!
+
+#![no_std]
+extern crate alloc;
+
+use core::mem::{self, transmute};
+
+use alloc::{format, string::String, vec, vec::Vec};
+use r_efi::{
+  efi,
+  hii::{self, PACKAGE_END},
+  protocols::hii_database::*,
+};
+use scroll::{ctx, Pread, Pwrite};
+
+/// HII Keyboard Package List
+/// Refer to UEFI spec version 2.10 section 33.3.1.2 which defines the generic header structure. This implementation
+/// only supports HII Keyboard Packages; other HII package types (or mixes) are not supported.
+#[derive(Debug, PartialEq, Eq)]
+pub struct HiiKeyboardPkgList {

```

Comment:
It seems like there is an opportunity to group these `Hii` related structures and implementation into a module named `hii` (like hii.rs) and then refer to them from this main library file as something like `hii::KeyboardPkgList`. That would keep the structures and associated implementation in one place and the other library functions in another place while reducing the overall size of this file.

Though the library is very "HII" centric so I could see preference to keep them in the main library file. Just a thought to separate things out some.